docs/docs/external-plugins.md

```diff
@@ -138,6 +152,48 @@ bash-3.2$ /path/to/myplugin.rb list /myplugin/foo ''
 ]
 ```
 
+### Prefetch
+
+`list` can prefetch other it's entries. The output would be an array of other entries, identical to the output from running the script in the previous examples.
+
+Additionally Wash supports _core entries_ that provide built-in functionality that plugins can take advantage of.
+
+_core entries_ can be used by returning an entry object with `type_id` set to the _core entries_ name surrounded by double underscores (`__core::entry__`) and the `name` field to identify the entry. You can also specify options for a _core entry_ using the `state` field containing serialized JSON of the options. If your plugin use _schemas_, then when using a _core entry_ you must still specify the _core entry_ as a child of the entry that lists it in the schema by it's `type_id`.
+
+* `volume::fs`: a representation of a filesystem that uses your entry's `exec` method to access it. _Options_:
```

Comment:
Nit: representation of your entry's filesystem that uses its `exec` method to access it? The current wording makes it seem more abstract than what it is (e.g. what filesystem is it representing?)